src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNITcpHandle.cs

```diff
@@ -454,7 +455,10 @@ private uint ReportTcpSNIError(string errorMessage)
 
         private uint ReportErrorAndReleasePacket(SNIPacket packet, string errorMessage)
         {
-            packet.Release();
+            if (packet != null)
+            {
+                packet.Release();                
```

Comment:
We still have a number of build/test infrastructures that use the same code and not all of them are building with the Roslyn toolset yet. Hopefully we will be able to allow it in the coming months.